RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFClusterECLCC.h

```diff
@@ -0,0 +1,297 @@
+
+     * Redistributions of source code must retain the above copyright
+       notice, this list of conditions and the following disclaimer.
+     * Redistributions in binary form must reproduce the above copyright
+       notice, this list of conditions and the following disclaimer in the
+       documentation and/or other materials provided with the distribution.
+     * Neither the name of Texas State University nor the names of its
+       contributors may be used to endorse or promote products derived from
+       this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  DISCLAIMED. IN NO EVENT SHALL TEXAS STATE UNIVERSITY BE LIABLE FOR ANY
+  DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+  Authors: Jayadharini Jaiganesh and Martin Burtscher
+
+  URL: The latest version of this code is available at
+  https://userweb.cs.txstate.edu/~burtscher/research/ECL-CC/.
+
+  Publication: This work is described in detail in the following paper.
+  Jayadharini Jaiganesh and Martin Burtscher. A High-Performance Connected
+  Components Implementation for GPUs. Proceedings of the 2018 ACM International
+  Symposium on High-Performance Parallel and Distributed Computing, pp. 92-104.
+  June 2018.
+*/
+
+/*
+ The code is modified for the specific use-case of generating topological clusters
+ for PFClustering. It is adapted to work with the Alpaka portability library.
+ 
+*/
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  /* intermediate pointer jumping */
+
+  ALPAKA_FN_ACC inline int representative(const int idx,
+                                          reco::PFClusteringVarsDeviceCollection::View pfClusteringVars) {
+    int curr = pfClusteringVars[idx].pfrh_topoId();
+    if (curr != idx) {
+      int next, prev = idx;
+      while (curr > (next = pfClusteringVars[curr].pfrh_topoId())) {
+        pfClusteringVars[prev].pfrh_topoId() = next;
+        prev = curr;
+        curr = next;
+      }
+    }
+    return curr;
+  }
+
+  // Initial step of ECL-CC. Uses ID of first neighbour in edgeList with a smaller ID
+  class ECLCCInit {
+  public:
+    template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
+    ALPAKA_FN_ACC void operator()(const TAcc& acc,
+                                  const reco::PFRecHitHostCollection::ConstView pfRecHits,
```

Comment:
You don't need to pass `ConstView` as `const`.
It is not wrong, but it doesn't make any difference, either.